app/Controllers/configureController.php

```diff
@@ -196,27 +196,49 @@ public function shortcutAction() {
 	 */
 	public function archivingAction() {
 		if (Minz_Request::isPost()) {
+			if (!$enableRetentionCountLimit = Minz_Request::paramBoolean('enable_retention_count_limit')) {
+				$retentionCountLimit = null;
+			} elseif (!$retentionCountLimit = Minz_Request::param('retention_count_limit')) {
+				$retentionCountLimit = FreshRSS_Feed::ARCHIVING_RETENTION_COUNT_LIMIT;
+			}
+			if ($enableRetentionPeriod = Minz_Request::paramBoolean('enable_retention_period')) {
+				$retentionPeriod = FreshRSS_Feed::ARCHIVING_RETENTION_PERIOD;
+				if (is_numeric(Minz_Request::param('retention_period_count')) && preg_match('/^PT?1[YMWDH]$/', Minz_Request::param('retention_period_unit'))) {
+					$retentionPeriod = str_replace(1, Minz_Request::param('retention_period_count'), Minz_Request::param('retention_period_unit'));
```

Comment:
Isn't there a problem with the hard-coded `str_replace(1` and the subject which can be `"P1..."` or `"PT1..."` according to the `preg_match` one line before?
It might be OK in that context to allow only one format.